wkz/device/mount.py

```diff
@@ -0,0 +1,155 @@
+    if "Mounted" not in mount_output:
+        raise FailedToMountDevice(
+            f"Output of mount command does not look as expected: {mount_output}. Expected to contain 'Mounted'."
+        )
+    path_start = mount_output.find("at")
+    mount_path = mount_output[path_start + 3 :]
+    return mount_path
+
+
+def _get_path_to_device(lsusb_output: str) -> str:
+    assert DEVICE_READY_STRING in str(lsusb_output)
+    list_of_lines = str(lsusb_output).split("\n")
+    path = None
+    for line in list_of_lines:
+        if DEVICE_READY_STRING in line:
+            bus_start = line.find("Bus") + 4
+            bus = line[bus_start : bus_start + 3]
+            device_start = line.find("Device") + 7
+            dev = line[device_start : device_start + 3]
+            path = f"/dev/bus/usb/{bus}/{dev}"
+            return path
+
+
+def _mount_device_using_gio(path: str) -> str:
+    return subprocess.check_output(["gio", "mount", "-d", path]).decode("utf-8").rstrip()
+
+
+def _mount_device_using_pmount(path: str) -> str:
+    subprocess.check_output(["pmount", path, "garmin"]).decode("utf-8")
+    # pmount does not return anything, assume command to be successful and return expected string
+    return "Mounted at /media/garmin"
+
+
+def garmin_device_connected() -> bool:
+    try:
+        lsusb = _get_lsusb_output()
+    except (FileNotFoundError, subprocess.CalledProcessError):
+        raise FailedToMountDevice("No 'lsusb' command available on your system.")
+    if "Garmin" in lsusb:
+        return True
+    else:
+        return False
+
+
+def _get_lsusb_output() -> str:
+    return subprocess.check_output("lsusb").decode("utf8")
+
+
+def _determine_type_and_mount(path: str) -> str:
+    device_type = _determine_device_type(path)
+    log.debug(f"device at path {path} is of type {device_type}")
+    try:
+        if device_type == "MTP":
+            return _mount_device_using_gio(path)
+        elif device_type == "BLOCK":
+            return _mount_device_using_pmount(path)
+    except subprocess.CalledProcessError as e:
+        raise FailedToMountDevice(f"Execution of mount command failed: {e}")
+
+
+def _determine_device_type(path: str) -> str:
+    log.debug(f"trying to determine device type for device at: {path}...")
+    device_tree = pyudev.Context()
+    if _is_of_type_mtp(device_tree, path):
+        return "MTP"
+    elif _is_of_type_block(device_tree, path):
+        return "BLOCK"
+    else:
+        raise FailedToMountDevice("Could not determine device type. Device is neither MTP nor BLOCK.")
+
+
+def _is_of_type_mtp(device_tree: pyudev.core.Context, path: str) -> bool:
+    devices = (
+        device_tree.list_devices(subsystem="usb").match_property("DEVNAME", path).match_property("ID_MTP_DEVICE", "1")
+    )
+    if len(list(devices)) > 0:
+        return True
+    else:
+        return False
+
+
+def _is_of_type_block(device_tree: pyudev.core.Context, path: str) -> bool:
```

Comment:
@gotiniens I changed the implementation of the mount mechanism quite a bit. However, I tried to stick to what we used to have in place for determining the device type as far as possible. As we have discussed earlier testing this piece of code is of course not trivial. So I would appreciate if you could spend a moment for testing this implementation with your device, before I merge it. 

You should be able to directly install the implementation of this PR using
```
pip install git+https://github.com/fgebhart/workoutizer.git@refs/pull/214/head#egg=workoutizer
```

Let me know if you encounter any issues.

Reply:
Note, that these changes also require you to use the simplified version of the `udev` rule as given here: https://github.com/fgebhart/workoutizer/pull/214/files#diff-aa7a481338e2c1cee33238469338ef1d478d3912ced41db1c35146b9e6521b1bR52